sycl/test-e2e/DeviceLib/built-ins/nan.cpp

```diff
@@ -57,13 +71,18 @@ int main() {
     }
   });
 
-  if (Queue.get_device().has(sycl::aspect::fp16))
-    check_nan<unsigned short, s::half>(Queue);
+  if (Queue.get_device().has(sycl::aspect::fp16)) {
+    check_scalar_nan<unsigned short, s::half>(Queue);
+    check_vec_nan<uint16_t, s::half>(Queue);
+  }
 
-  check_nan<unsigned int, float>(Queue);
+  check_scalar_nan<unsigned int, float>(Queue);
+  check_vec_nan<uint32_t, float>(Queue);
   if (Queue.get_device().has(sycl::aspect::fp64)) {
-    check_nan<unsigned long, double>(Queue);
-    check_nan<unsigned long long, double>(Queue);
+    check_scalar_nan<unsigned long, long_float_t>(Queue);
+    check_scalar_nan<unsigned long long, double>(Queue);
+    check_vec_nan<uint64_t, double>(Queue);
+    check_vec_nan<unsigned long long, double>(Queue);
   }
   return 0;
```

Comment:
Unrelated to this line,

Generated file has, e.g., 
```
__SYCL_DEPRECATED("SYCL builtin functions with raw pointer arguments have been deprecated. Please use multi_ptr.")
inline double lgamma_r(double a0, int* a1) __NOEXC {
  return __sycl_std::__invoke_lgamma_r<double>(a0, a1);
}
```

But current implementation does not accept raw pointer:

```
template <typename T, typename T2>
std::enable_if_t<
    detail::is_svgenfloat<T>::value && detail::is_genintptr<T2>::value, T>
lgamma_r(T x, T2 signp) __NOEXC {
  detail::check_vector_size<T, T2>();
  return __sycl_std::__invoke_lgamma_r<T>(x, signp);
}
```
It should be probably some kind of regression because we introduce old API with this patch.

Reply:
I wish that was the case, but `is_genintptr` is defined as

```c++
template <typename T>
using is_genintptr = std::bool_constant<
    is_pointer<T>::value && is_genint<remove_pointer_t<T>>::value &&
    is_address_space_compliant<T, gvl::nonconst_address_space_list>::value>;
```
and
```c++
...

template <typename T> struct is_pointer_impl<T *> : std::true_type {};

...

template <typename T>
struct is_pointer : is_pointer_impl<std::remove_cv_t<T>> {};
```

so for this case `int *` is considered `genintptr` by our implementation. As such I added (now deprecated) overloads that take raw pointers to avoid breaking those without deprecation. Hopefully we can remove them for good in the near future.

Reply:
Apologies, indeed